cloud-nio/cloud-nio-spi/src/main/scala/cloud/nio/spi/CloudNioPath.scala

```diff
@@ -17,7 +17,7 @@ object CloudNioPath {
   }
 }
 
-class CloudNioPath(filesystem: CloudNioFileSystem, private[spi] val unixPath: UnixPath) extends Path {
+class CloudNioPath(val filesystem: CloudNioFileSystem, private[spi] val unixPath: UnixPath) extends Path {
```

Comment:
On first glance this smells funny to me.

1) I'm wondering if this particular encapsulation should be broken just to access the martha configuration.
2) That said, the abstract `java.nio.file.FileSystem` is also available via [`java.nio.Path.getFileSystem()`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#getFileSystem--).

Reply:
ToL: `java.nio.file.Path` lets you [access its filesystem](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#getFileSystem--) so the parallel with the `CloudNioPath` doesn't shock me

Reply:
I have reverted that change. I must have missed `java.nio.Path.getFileSystem()` somehow! I am using `drsPath.getFileSystem.provider.asInstanceOf[DrsCloudNioFileSystemProvider]` as suggested.